File: client/src/entities/Product.ts
import PageResponse from "./PageResponse";
import { UserData } from "./User";

export interface ProductImageModel {
  productImageId: number;
  productImage: string;
}

export interface ProductModel {
  productId: number;
  category: string;
  productCondition: string;
  availability: string;
  productName: string;
  price: number;
  brand: string;
  description: string;
  timestamp: string;
  productImages: ProductImageModel[];
  user: UserData;
}

export default interface ProductResponse {
  productModels: ProductModel[];
  pageResponse: PageResponse;
}

File: server/src/main/java/com/facebook/server/entity/Product.java
package com.facebook.server.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

import com.facebook.server.entity.constants.Availability;
import com.facebook.server.entity.constants.Category;
import com.facebook.server.entity.constants.ProductCondition;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "products")
public class Product extends Timestamp {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long productId;
    @Enumerated(EnumType.STRING)
    private Category category;
    @Enumerated(EnumType.STRING)
    private ProductCondition productCondition;
    @Enumerated(EnumType.STRING)
    private Availability availability;
    private String productName;
    private Double price;
    private String brand;
    private String description;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<ProductImage> productImages = new ArrayList<>();
}
